### PR TITLE
Saving epic tests

### PR DESCRIPTION
**Pull request by jdetle**
Put this up so someone can check it out and help me debug. Currently actions aren't getting pushed that should be getting emitted, please help if you can. Seriously confused as to why line 66 is working and line 80 isn't. Thanks for any help that people are willing to give :)

### CODE REVIEW

**Comment by lgeiger** [comment]
@jdetle I think you are missing to call the `done()` callback on the in your tests. I suppose line 80 only works because it's synchronous.

I can push a fix, but it'll probably make both tests fail :wink:

**Comment by jdetle** [comment]
So one issue was that in my observer, I wasn't actually passing any args to actionBuffer.push, which was definitely a problem. But I'll look into that too, thanks.

**Comment by rgbkrk** [comment]
Looks like this is passing now - still need folks to check on stuff or is it ready to merge?

**Comment by jdetle** [comment]
I want you to check what I added to the saving epic, but other than that I think its ready to ship.

File: src/notebook/epics/saving.js [hunk 1]
```diff
@@ -32,15 +33,21 @@ export const saveEpic = actions =>
                 value.delete('inputHidden').delete('outputHidden').delete('status')))),
           null,
           1))
+        .catch(error => {
+          const input$ = Observable.of({ type: 'SAVING_ERROR' });
```
Comment (rgbkrk) [inline src/notebook/epics/saving.js]: You can just return `Observable.of` here. I would also include the error similar to our other payloads so this is easier to debug/deal with.

**Review by rgbkrk — APPROVED** [review]
One thing to change if you like, otherwise it's good to go.

File: src/notebook/epics/saving.js (hunk 1 above)
Comment (jdetle) [reply src/notebook/epics/saving.js]: What would that error be? My first thought would be to create a new error action?

Comment (rgbkrk) [reply src/notebook/epics/saving.js]:
``` js
.catch(error => { // <-- This one
```

Comment (jdetle) [reply src/notebook/epics/saving.js]: Oh! Right. Silly me.

Comment (rgbkrk) [reply src/notebook/epics/saving.js]: You would make this:

``` js
Observable.of({ type: 'ERROR_SAVING', payload: error })
```

**Comment by jdetle** [comment]
90% is so close I can almost taste it.

**Pull request merged**